util/bloom/generic.go

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2019-2021 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package bloom
+
+// GenericFilter is the interface for either bloom.Filter or bloom.XorFilter
+type GenericFilter interface {
```

Comment:
I'm not opposing to having this interface, because it's effectively aligns with our usage of the bloom filter.
However, note that not all the bloom filter types would align with that; in particular, some bloom filters let you set and delete entries as well as continuously modify the bloom filter.